### PR TITLE
Feature/add

### PR DESCRIPTION
**Pull request by davidx**
This is the initial version of the github action for datadog.
This is a plugin specifically made for integration within the parsleyhealth cicd system. 

tested, triggered successfully by using the following yaml:

```
on: [push]

jobs:
  deploy_job:
    runs-on: ubuntu-latest
    name: A test datadog
    steps:
      - name: github_action_datadog
        id: hello
        uses: parsleyhealth/github-action-datadog@1585616101
        with:
          datadog_api_key: ${{ secrets.DATADOG_API_KEY_STAGING }}
          parsley_componentname: cicd
          parsley_environment: "staging"

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,21 @@
-# github-action-datadog
\ No newline at end of file
+# Hello world javascript action
```
Comment (ChairmanTubeAmp) [inline README.md:1]: Please replace with some instructions on use of the plugin.

File: action.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: 'Notification event to datadog'
+description: 'Notify datadog with events'
+branding:
+  icon: message-square
+  color: orange
+inputs:
```
Comment (ChairmanTubeAmp) [inline action.yml:6]: Michael suggests a style change here replacing `_` with `-` in the input variable names.

https://parsleyhealth.slack.com/archives/CUJCL6XK5/p1585603489074800

File: index.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
+
+const core = require("@actions/core");
+const github = require("@actions/github");
+const httpm = require("@actions/http-client");
+
+const run = async () => {
+    const datadog_api_key = core.getInput("datadog_api_key");
+    const datadog_uri_prefix = "https://api.datadoghq.com/api/v1/series?api_key="
+    const datadog_metric_uri = datadog_uri_prefix + datadog_api_key
+    const datadog_event_uri = datadog_uri_prefix + datadog_api_key
```
Comment (ChairmanTubeAmp) [inline index.js]: these two variables seem to have identical data - maybe delete one and rename the other datadog_uri?

File: index.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    const parsley_componentname = core.getInput("parsley_componentname");
+    const parsley_environment = core.getInput("parsley_environment");
+
+    let payload = await github.context.payload;
+
+    const current_time = parseInt(Math.round((new Date()).getTime() / 1000));
+    const reponame = payload.repository.full_name;
+    const ref_path = payload.ref;
+    const branchname = ref_path.split('/').pop();
+    const gitsha = payload.after; // confirm this is correct.
```
Comment (ChairmanTubeAmp) [inline index.js]: is this comment still relevant? if not, please remove

File: index.js [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    const gitauthor = payload.commits[0]['author']['username'];
+    const head_commit_timestamp = Date.parse(payload.head_commit['timestamp']);
+    const last_commit_epoch = parseInt(Math.round(head_commit_timestamp / 1000));
+    const lead_time = (current_time - last_commit_epoch);
+
+    const event_title = "Deploy event for " + parsley_componentname + " in env: " + parsley_environment;
+    const event_text = "Deploy event for " + parsley_componentname + " in env: " + parsley_environment;
+    const event_priority = "normal";
+
+    const http = new httpm.HttpClient("http-client-tests");
```
Comment (ChairmanTubeAmp) [inline index.js]: "http-client-tests" is the user agent that the example used. We can change to something more appropriate for Parsley.

File: index.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    const parsley_tags = [
+        "environment:" + parsley_environment,
+        "componentname:" + parsley_componentname,
+        "reponame:" + reponame,
+        "branchname:" + branchname,
+        "gitsha:" + gitsha,
+        "gitauthor:" + gitauthor
+    ];
+
+    let datadog_lead_time_metric_payload = JSON.stringify({
```
Comment (ChairmanTubeAmp) [inline index.js:40]: we lost "units" - the unit of measure tag. It was set to seconds.

File: index.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+              points: [[current_time, lead_time]],
+              type: "count",
+              interval: 20,
+              host: "cicd.parsleyhealth.com",
+              tags: parsley_tags
+            }
+            ]
+    });
+    let datadog_metric_payload = JSON.stringify({
+            series: [
```
Comment (ChairmanTubeAmp) [inline index.js:54]: here "units" was set to boolean

File: package.json [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/parsleyhealth/github-action-datadog.git"
+  },
+  "keywords": [],
+  "author": "",
```
Comment (ChairmanTubeAmp) [inline package.json]: you should add yourself as an author

**Review by ChairmanTubeAmp — CHANGES_REQUESTED** [review]

File: index.js (hunk 3 above)
Comment (davidx) [reply index.js]: outdated

File: index.js (hunk 5 above)
Comment (davidx) [reply index.js]: like what ?

File: action.yml (hunk 2 above)
Comment (davidx) [reply action.yml:6]: this is a builtin, the icon is named that.

Comment (davidx) [reply action.yml:6]: Ah i see, Ok i fixed in inputs and outputs. thx! @ChairmanTubeAmp

**Review by ChairmanTubeAmp — APPROVED** [review]

**Pull request merged**